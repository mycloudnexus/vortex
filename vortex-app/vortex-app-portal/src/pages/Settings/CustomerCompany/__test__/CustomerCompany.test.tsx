import { fireEvent, render, waitFor } from '@testing-library/react'

import { MemoryRouter } from 'react-router-dom'
import CustomerCompany from '..'
import { ReactElement } from 'react'
import { QueryClient, QueryClientProvider } from 'react-query'

global.matchMedia = jest.fn().mockImplementation((query) => ({
  matches: false,
  media: query,
  addListener: jest.fn(),
  removeListener: jest.fn()
}))

jest.mock('@/assets/icon/customer-company.svg', () => ({
  ReactComponent: () => <svg data-testid='customer-company' />
}))
jest.mock('@/assets/icon/customer-company-empty.svg', () => ({
  ReactComponent: () => <svg data-testid='customer-company-empty' />
}))
jest.mock('@/assets/icon/status.svg', () => ({
  ReactComponent: () => <svg data-testid='status' />
}))
jest.mock('@/assets/icon/close-circle.svg', () => ({
  ReactComponent: () => <svg data-testid='close-circle' />
}))
jest.mock('@/assets/icon/warning-circle.svg', () => ({
  ReactComponent: () => <svg data-testid='warning-circle' />
}))
jest.mock('@/assets/icon/info.svg', () => ({
  ReactComponent: () => <svg data-testid='info' />
}))

jest.mock('@/hooks/company', () => ({
  useGetCompanyList: jest.fn(),
  useAddOrganization: jest.fn(),
  useUpdateOrganization: jest.fn()
}))

const mockedUseGetCompanyList = require('@/hooks/company').useGetCompanyList
const mockedUseAddOrganization = require('@/hooks/company').useAddOrganization
const mockedUseUpdateOrganization = require('@/hooks/company').useUpdateOrganization
const dummyData = {
  code: 200,
  message: 'OK',
  data: {
    data: [
      {
        name: 'helloworld123',
        id: 'org_D4ES55BSeeAHystq',
        display_name: 'hello_world',
        metadata: {
          loginType: 'undefined',
          status: 'ACTIVE',
          type: 'CUSTOMER'
        }
      },
      {
        name: 'qetesting',
        id: 'org_AeltT2tTQsOFNvCC',
        display_name: 'QE Testing'
      }
    ],
    total: 2,
    page: 0,
    size: 20
  }
}
describe('Customer Company Page', () => {
  let component: ReactElement
  const queryClient = new QueryClient()
  jest.spyOn(queryClient, 'setQueryData')
  beforeEach(() => {
    mockedUseAddOrganization.mockReturnValue({
      mutate: jest.fn()
    })

    mockedUseUpdateOrganization.mockReturnValue({
      mutate: jest.fn()
    })

    component = (
      <QueryClientProvider client={queryClient}>
        <MemoryRouter>
          <CustomerCompany />
        </MemoryRouter>
      </QueryClientProvider>
    )
  })

  afterEach(() => {
    jest.clearAllMocks()
  })
  it('should render table columns', () => {
    mockedUseGetCompanyList.mockReturnValue({
      data: [],
      isLoading: false,
      isError: false
    })
    const { getByText } = render(component)

    expect(getByText('Name')).toBeInTheDocument()
    expect(getByText('ID')).toBeInTheDocument()
    expect(getByText('Short name & URL')).toBeInTheDocument()
    expect(getByText('Status')).toBeInTheDocument()
    expect(getByText('Action')).toBeInTheDocument()
  })

  it('should display inactive company data in table rows', () => {
    mockedUseGetCompanyList.mockReturnValue({
      data: dummyData,
      isLoading: false,
      isError: false
    })
    const { getByText, baseElement } = render(component)
    expect(getByText('QE Testing')).toBeInTheDocument()
    expect(getByText('qetesting')).toBeInTheDocument()
    expect(getByText('org_AeltT2tTQsOFNvCC')).toBeInTheDocument()
    expect(getByText('Inactive')).toBeInTheDocument()
    expect(baseElement).toMatchSnapshot()
  })

  it('should call handleDeactivate when clicking Deactivate button', async () => {
    //will update this test
    const { getByTestId, getByText } = render(component)
    fireEvent.click(getByTestId('handle-deactivate'))
    await waitFor(() => {
      expect(getByTestId('deactivate-modal')).toBeInTheDocument()
    })
    const submitButton = getByText('Yes, continue')
    fireEvent.click(submitButton)
  })

  it('should call handleActivate when clicking Activate button', () => {
    //will update this test
    const { getByTestId } = render(component)
    fireEvent.click(getByTestId('handle-activate'))
  })
  it('add form renders correctly and cancel submit', async () => {
    //Will update this test
    const { getByTestId, getByText, queryByTestId } = render(component)
    const button = getByTestId('add-button')
    expect(button).toBeInTheDocument()
    fireEvent.click(button)
    await waitFor(() => {
      expect(getByTestId('add-modal')).toBeInTheDocument()
    })
    const submitButton = getByText('Cancel')
    fireEvent.click(submitButton)
    await waitFor(() => {
      expect(queryByTestId('add-modal')).not.toBeInTheDocument()
    })
  })

  // Helper function to handle form submission
  const handleFormSubmitForUpdate = async ({ companyName }: { companyName: string }) => {
    const { getByTestId, getByLabelText, getByText } = render(component)

    const button = getByTestId('handle-modify')
    expect(button).toBeInTheDocument()

    fireEvent.click(button)

    await waitFor(() => {
      expect(getByTestId('update-modal')).toBeInTheDocument()
    })

    fireEvent.change(getByLabelText(/Customer company name/i), { target: { value: companyName } })

    const submitButton = getByText('OK')
    fireEvent.click(submitButton)

    return { getByTestId, getByLabelText, getByText }
  }

  it('should call useUpdateOrganization mutation on form submission', async () => {
    const mockMutate = jest.fn((_data, { onSuccess }) => {
      onSuccess({
        data: {
          id: 'org_D4ES55BSeeAHystq',
          display_name: 'My Company'
        }
      })
    })

    mockedUseUpdateOrganization.mockReturnValue({
      mutate: mockMutate
    })

    await handleFormSubmitForUpdate({ companyName: 'My Company' })

    await waitFor(() => {
      expect(mockMutate).toHaveBeenCalledWith(
        expect.objectContaining({
          id: 'org_D4ES55BSeeAHystq',
          request_body: {
            display_name: 'My Company'
          }
        }),
        expect.objectContaining({
          onSuccess: expect.any(Function),
          onError: expect.any(Function)
        })
      )
    })

    await waitFor(() => {
      expect(mockMutate).toHaveBeenCalledTimes(1)
    })
  })

  it('should handle useUpdateOrganization onError callback', async () => {
    const mockMutate = jest.fn((_data, { onError }) => {
      const mockError = new Error('Network error')
      onError(mockError)
    })

    const consoleLogSpy = jest.spyOn(console, 'log').mockImplementation(() => {})

    mockedUseUpdateOrganization.mockReturnValue({
      mutate: mockMutate
    })

    await handleFormSubmitForUpdate({ companyName: 'My Company' })

    await waitFor(() => {
      expect(mockMutate).toHaveBeenCalledWith(
        expect.objectContaining({
          id: 'org_D4ES55BSeeAHystq',
          request_body: {
            display_name: 'My Company'
          }
        }),
        expect.objectContaining({
          onSuccess: expect.any(Function),
          onError: expect.any(Function)
        })
      )
    })

    await waitFor(() => {
      expect(consoleLogSpy).toHaveBeenCalledWith(expect.any(Error), 'error update')
    })

    expect(mockMutate).toHaveBeenCalledTimes(1)

    consoleLogSpy.mockRestore()
  })

  it('form update renders correctly and cancel submit', async () => {
    const { getByTestId, getByText, queryByTestId } = render(component)
    const button = getByTestId('handle-modify')
    expect(button).toBeInTheDocument()
    fireEvent.click(button)
    await waitFor(() => {
      expect(getByTestId('update-modal')).toBeInTheDocument()
    })
    const cancelButton = getByText('Cancel')
    fireEvent.click(cancelButton)
    await waitFor(() => {
      expect(queryByTestId('update-modal')).not.toBeInTheDocument()
    })
  })

  // Helper function to handle form submission
  const handleFormSubmit = async ({ companyName, companyUrl }: { companyName: string; companyUrl: string }) => {
    const { getByTestId, getByLabelText, getByText } = render(component)

    fireEvent.click(getByTestId('add-button'))

    await waitFor(() => {
      expect(getByTestId('add-modal')).toBeInTheDocument()
    })

    fireEvent.change(getByLabelText(/Customer company name/i), {
      target: { value: companyName }
    })
    fireEvent.change(getByLabelText(/Customer company URL short name/i), {
      target: { value: companyUrl }
    })
    fireEvent.click(getByText('OK'))

    return { getByTestId, getByLabelText, getByText }
  }

  it('should call useAddOrganization mutation on form submission', async () => {
    const mockMutate = jest.fn((data, { onSuccess }) => {
      onSuccess({
        data: {
          display_name: data.display_name,
          name: data.name.toLowerCase()
        }
      })
    })

    mockedUseAddOrganization.mockReturnValue({
      mutate: mockMutate
    })

    await handleFormSubmit({ companyName: 'My New Company', companyUrl: 'abc' })

    await waitFor(() => {
      expect(mockMutate).toHaveBeenCalledWith(
        expect.objectContaining({
          display_name: 'My New Company',
          name: 'abc'
        }),
        expect.objectContaining({
          onSuccess: expect.any(Function),
          onError: expect.any(Function)
        })
      )
    })
  })

  it('should handle useAddOrganization onError callback', async () => {
    const mockMutate = jest.fn((_data, { onError }) => {
      const mockError = new Error('Network error')
      onError(mockError)
    })

    const consoleLogSpy = jest.spyOn(console, 'log').mockImplementation(() => {})
    mockedUseAddOrganization.mockReturnValue({
      mutate: mockMutate
    })

    await handleFormSubmit({ companyName: 'My New Company', companyUrl: 'abc' })

    await waitFor(() => {
      expect(mockMutate).toHaveBeenCalledWith(
        expect.objectContaining({
          display_name: 'My New Company',
          name: 'abc'
        }),
        expect.objectContaining({
          onSuccess: expect.any(Function),
          onError: expect.any(Function)
        })
      )
    })

    await waitFor(() => {
      expect(consoleLogSpy).toHaveBeenCalledWith(expect.any(Error), 'error adding')
    })
    expect(mockMutate).toHaveBeenCalledTimes(1)
    consoleLogSpy.mockRestore()
  })
})
