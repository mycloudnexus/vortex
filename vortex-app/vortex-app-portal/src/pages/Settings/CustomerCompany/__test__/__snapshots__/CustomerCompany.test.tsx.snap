// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Customer Company Page form renders correctly and modal opens 1`] = `
<body>
  <div>
    <div
      class="ant-flex css-dev-only-do-not-override-ni1kz0 ant-flex-align-center ant-flex-justify-start ant-flex-vertical"
      style="width: 70%; margin: 0px auto; gap: 20px;"
    >
      <div
        class="ant-flex css-dev-only-do-not-override-ni1kz0 ant-flex-align-center ant-flex-justify-space-between"
        style="width: 100%;"
      >
        <div
          class="ant-flex css-dev-only-do-not-override-ni1kz0 ant-flex-align-center"
          style="gap: 15px;"
        >
          <div
            class="sc-fAUdSK dfFYLE"
          >
            <svg
              data-testid="customer-company"
            />
          </div>
          <h3
            class="ant-typography css-dev-only-do-not-override-ni1kz0"
            style="margin: 0px;"
          >
            Customer Company
          </h3>
        </div>
        <button
          class="ant-btn css-dev-only-do-not-override-ni1kz0 ant-btn-default ant-btn-color-default ant-btn-variant-outlined sc-egkSDF kCIHkZ"
          data-testid="add-button"
          type="button"
        >
          <span>
            Add customer company
          </span>
        </button>
      </div>
      <div
        class="ant-table-wrapper sc-dntaoT jPeUFk css-dev-only-do-not-override-ni1kz0"
        style="width: 100%;"
      >
        <div
          class="ant-spin-nested-loading css-dev-only-do-not-override-ni1kz0"
        >
          <div
            class="ant-spin-container"
          >
            <div
              class="ant-table ant-table-empty css-dev-only-do-not-override-ni1kz0 ant-table-fixed-header"
            >
              <div
                class="ant-table-container"
              >
                <div
                  class="ant-table-header"
                  style="overflow: hidden;"
                >
                  <table
                    style="table-layout: fixed;"
                  >
                    <colgroup />
                    <thead
                      class="ant-table-thead"
                    >
                      <tr>
                        <th
                          class="ant-table-cell"
                          scope="col"
                        >
                          Name
                        </th>
                        <th
                          class="ant-table-cell"
                          scope="col"
                        >
                          ID
                        </th>
                        <th
                          class="ant-table-cell"
                          scope="col"
                        >
                          Short name & URL
                        </th>
                        <th
                          class="ant-table-cell"
                          scope="col"
                        >
                          Status
                        </th>
                        <th
                          class="ant-table-cell"
                          scope="col"
                        >
                          Action
                        </th>
                      </tr>
                    </thead>
                  </table>
                </div>
                <div
                  class="ant-table-body"
                  style="overflow-y: auto; max-height: 560px;"
                >
                  <table
                    style="table-layout: fixed;"
                  >
                    <colgroup />
                    <tbody
                      class="ant-table-tbody"
                    >
                      <tr
                        aria-hidden="true"
                        class="ant-table-measure-row"
                        style="height: 0px; font-size: 0px;"
                      >
                        <td
                          style="padding: 0px; border: 0px; height: 0px;"
                        >
                          <div
                            style="height: 0px; overflow: hidden;"
                          >
                             
                          </div>
                        </td>
                        <td
                          style="padding: 0px; border: 0px; height: 0px;"
                        >
                          <div
                            style="height: 0px; overflow: hidden;"
                          >
                             
                          </div>
                        </td>
                        <td
                          style="padding: 0px; border: 0px; height: 0px;"
                        >
                          <div
                            style="height: 0px; overflow: hidden;"
                          >
                             
                          </div>
                        </td>
                        <td
                          style="padding: 0px; border: 0px; height: 0px;"
                        >
                          <div
                            style="height: 0px; overflow: hidden;"
                          >
                             
                          </div>
                        </td>
                        <td
                          style="padding: 0px; border: 0px; height: 0px;"
                        >
                          <div
                            style="height: 0px; overflow: hidden;"
                          >
                             
                          </div>
                        </td>
                      </tr>
                      <tr
                        class="ant-table-placeholder"
                      >
                        <td
                          class="ant-table-cell"
                          colspan="5"
                        >
                          <div
                            class="ant-flex css-dev-only-do-not-override-ni1kz0 ant-flex-align-center ant-flex-justify-center ant-flex-vertical"
                          >
                            <svg
                              data-testid="customer-company-empty"
                            />
                            <span
                              style="font-size: 16px; font-weight: 500; color: rgb(0, 0, 0); font-style: normal;"
                            >
                              No customer company
                            </span>
                          </div>
                        </td>
                      </tr>
                    </tbody>
                  </table>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div>
    <div
      class="ant-modal-root css-dev-only-do-not-override-ni1kz0"
      data-testid="add-modal"
    >
      <div
        class="ant-modal-mask"
      />
      <div
        class="ant-modal-wrap ant-modal-centered"
        tabindex="-1"
      >
        <div
          aria-labelledby="test-id"
          aria-modal="true"
          class="ant-modal css-dev-only-do-not-override-ni1kz0 sc-blHHSb jFpWO ant-zoom-appear ant-zoom-appear-prepare ant-zoom"
          role="dialog"
          style="width: 520px;"
        >
          <div
            style="outline: none;"
            tabindex="0"
          >
            <div
              class="ant-modal-content"
            >
              <button
                aria-label="Close"
                class="ant-modal-close"
                type="button"
              >
                <span
                  class="ant-modal-close-x"
                >
                  <span
                    aria-label="close"
                    class="anticon anticon-close ant-modal-close-icon"
                    role="img"
                  >
                    <svg
                      aria-hidden="true"
                      data-icon="close"
                      fill="currentColor"
                      fill-rule="evenodd"
                      focusable="false"
                      height="1em"
                      viewBox="64 64 896 896"
                      width="1em"
                    >
                      <path
                        d="M799.86 166.31c.02 0 .04.02.08.06l57.69 57.7c.04.03.05.05.06.08a.12.12 0 010 .06c0 .03-.02.05-.06.09L569.93 512l287.7 287.7c.04.04.05.06.06.09a.12.12 0 010 .07c0 .02-.02.04-.06.08l-57.7 57.69c-.03.04-.05.05-.07.06a.12.12 0 01-.07 0c-.03 0-.05-.02-.09-.06L512 569.93l-287.7 287.7c-.04.04-.06.05-.09.06a.12.12 0 01-.07 0c-.02 0-.04-.02-.08-.06l-57.69-57.7c-.04-.03-.05-.05-.06-.07a.12.12 0 010-.07c0-.03.02-.05.06-.09L454.07 512l-287.7-287.7c-.04-.04-.05-.06-.06-.09a.12.12 0 010-.07c0-.02.02-.04.06-.08l57.7-57.69c.03-.04.05-.05.07-.06a.12.12 0 01.07 0c.03 0 .05.02.09.06L512 454.07l287.7-287.7c.04-.04.06-.05.09-.06a.12.12 0 01.07 0z"
                      />
                    </svg>
                  </span>
                </span>
              </button>
              <div
                class="ant-modal-header"
              >
                <div
                  class="ant-modal-title"
                  id="test-id"
                >
                  Add customer company
                </div>
              </div>
              <div
                class="ant-modal-body"
              >
                <form
                  class="ant-form ant-form-vertical css-dev-only-do-not-override-ni1kz0"
                  id="add_customer_company"
                  style="padding: 20px;"
                >
                  <div
                    class="ant-form-item css-dev-only-do-not-override-ni1kz0 ant-form-item-is-validating"
                  >
                    <div
                      class="ant-row ant-form-item-row css-dev-only-do-not-override-ni1kz0"
                    >
                      <div
                        class="ant-col ant-form-item-label css-dev-only-do-not-override-ni1kz0"
                      >
                        <label
                          class=""
                          for="add_customer_company_display_name"
                          title=""
                        >
                          <div
                            class="ant-flex css-dev-only-do-not-override-ni1kz0 ant-flex-align-center"
                            style="gap: 5px;"
                          >
                            Customer company name
                            <span
                              style="color: red;"
                            >
                              *
                            </span>
                          </div>
                        </label>
                      </div>
                      <div
                        class="ant-col ant-form-item-control css-dev-only-do-not-override-ni1kz0"
                      >
                        <div
                          class="ant-form-item-control-input"
                        >
                          <div
                            class="ant-form-item-control-input-content"
                          >
                            <input
                              class="ant-input css-dev-only-do-not-override-ni1kz0 ant-input-outlined ant-input-status-validating sc-gtLWhw jtrSU"
                              data-testid="customer-name"
                              id="add_customer_company_display_name"
                              maxlength="255"
                              placeholder="Please enter"
                              type="text"
                              value="My Company"
                            />
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                  <div
                    class="ant-form-item css-dev-only-do-not-override-ni1kz0 ant-form-item-is-validating"
                  >
                    <div
                      class="ant-row ant-form-item-row css-dev-only-do-not-override-ni1kz0"
                    >
                      <div
                        class="ant-col ant-form-item-label css-dev-only-do-not-override-ni1kz0"
                      >
                        <label
                          class=""
                          for="add_customer_company_name"
                          title=""
                        >
                          <div
                            class="ant-flex css-dev-only-do-not-override-ni1kz0 ant-flex-align-center"
                            style="gap: 5px;"
                          >
                            Customer company URL short name
                            <span
                              data-testid="mocked-svg"
                              style="display: flex; justify-content: center;"
                            >
                              <svg
                                data-testid="info"
                              />
                            </span>
                            <span
                              style="color: red;"
                            >
                              *
                            </span>
                          </div>
                        </label>
                      </div>
                      <div
                        class="ant-col ant-form-item-control css-dev-only-do-not-override-ni1kz0"
                      >
                        <div
                          class="ant-form-item-control-input"
                        >
                          <div
                            class="ant-form-item-control-input-content"
                          >
                            <input
                              aria-describedby="add_customer_company_name_extra"
                              class="ant-input css-dev-only-do-not-override-ni1kz0 ant-input-outlined ant-input-status-validating sc-gtLWhw jtrSU"
                              id="add_customer_company_name"
                              maxlength="20"
                              placeholder="Please enter"
                              type="text"
                              value="MC"
                            />
                          </div>
                        </div>
                        <div
                          class="ant-form-item-extra"
                          id="add_customer_company_name_extra"
                        >
                          Only lower case letters, numbers, - , allowed. No more than 20 characters in total
                        </div>
                      </div>
                    </div>
                  </div>
                </form>
              </div>
              <div
                class="ant-modal-footer"
              >
                <button
                  class="ant-btn css-dev-only-do-not-override-ni1kz0 ant-btn-default ant-btn-color-default ant-btn-variant-outlined"
                  type="button"
                >
                  <span>
                    Cancel
                  </span>
                </button>
                <button
                  class="ant-btn css-dev-only-do-not-override-ni1kz0 ant-btn-primary ant-btn-color-primary ant-btn-variant-solid"
                  type="button"
                >
                  <span>
                    OK
                  </span>
                </button>
              </div>
            </div>
          </div>
          <div
            style="width: 0px; height: 0px; overflow: hidden; outline: none;"
            tabindex="0"
          />
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`Customer Company Page should display inactive company data in table rows 1`] = `
<body>
  <div>
    <div
      class="ant-flex css-dev-only-do-not-override-ni1kz0 ant-flex-align-center ant-flex-justify-start ant-flex-vertical"
      style="width: 70%; margin: 0px auto; gap: 20px;"
    >
      <div
        class="ant-flex css-dev-only-do-not-override-ni1kz0 ant-flex-align-center ant-flex-justify-space-between"
        style="width: 100%;"
      >
        <div
          class="ant-flex css-dev-only-do-not-override-ni1kz0 ant-flex-align-center"
          style="gap: 15px;"
        >
          <div
            class="sc-fAUdSK dfFYLE"
          >
            <svg
              data-testid="customer-company"
            />
          </div>
          <h3
            class="ant-typography css-dev-only-do-not-override-ni1kz0"
            style="margin: 0px;"
          >
            Customer Company
          </h3>
        </div>
        <button
          class="ant-btn css-dev-only-do-not-override-ni1kz0 ant-btn-default ant-btn-color-default ant-btn-variant-outlined sc-egkSDF kCIHkZ"
          data-testid="add-button"
          type="button"
        >
          <span>
            Add customer company
          </span>
        </button>
      </div>
      <div
        class="ant-table-wrapper sc-dntaoT jPeUFk css-dev-only-do-not-override-ni1kz0"
        style="width: 100%;"
      >
        <div
          class="ant-spin-nested-loading css-dev-only-do-not-override-ni1kz0"
        >
          <div
            class="ant-spin-container"
          >
            <div
              class="ant-table css-dev-only-do-not-override-ni1kz0 ant-table-fixed-header"
            >
              <div
                class="ant-table-container"
              >
                <div
                  class="ant-table-header"
                  style="overflow: hidden;"
                >
                  <table
                    style="table-layout: fixed; visibility: hidden;"
                  >
                    <colgroup />
                    <thead
                      class="ant-table-thead"
                    >
                      <tr>
                        <th
                          class="ant-table-cell"
                          scope="col"
                        >
                          Name
                        </th>
                        <th
                          class="ant-table-cell"
                          scope="col"
                        >
                          ID
                        </th>
                        <th
                          class="ant-table-cell"
                          scope="col"
                        >
                          Short name & URL
                        </th>
                        <th
                          class="ant-table-cell"
                          scope="col"
                        >
                          Status
                        </th>
                        <th
                          class="ant-table-cell"
                          scope="col"
                        >
                          Action
                        </th>
                      </tr>
                    </thead>
                  </table>
                </div>
                <div
                  class="ant-table-body"
                  style="overflow-y: scroll; max-height: 560px;"
                >
                  <table
                    style="table-layout: fixed;"
                  >
                    <colgroup />
                    <tbody
                      class="ant-table-tbody"
                    >
                      <tr
                        aria-hidden="true"
                        class="ant-table-measure-row"
                        style="height: 0px; font-size: 0px;"
                      >
                        <td
                          style="padding: 0px; border: 0px; height: 0px;"
                        >
                          <div
                            style="height: 0px; overflow: hidden;"
                          >
                             
                          </div>
                        </td>
                        <td
                          style="padding: 0px; border: 0px; height: 0px;"
                        >
                          <div
                            style="height: 0px; overflow: hidden;"
                          >
                             
                          </div>
                        </td>
                        <td
                          style="padding: 0px; border: 0px; height: 0px;"
                        >
                          <div
                            style="height: 0px; overflow: hidden;"
                          >
                             
                          </div>
                        </td>
                        <td
                          style="padding: 0px; border: 0px; height: 0px;"
                        >
                          <div
                            style="height: 0px; overflow: hidden;"
                          >
                             
                          </div>
                        </td>
                        <td
                          style="padding: 0px; border: 0px; height: 0px;"
                        >
                          <div
                            style="height: 0px; overflow: hidden;"
                          >
                             
                          </div>
                        </td>
                      </tr>
                      <tr
                        class="ant-table-row ant-table-row-level-0"
                        data-row-key="org_D4ES55BSeeAHystq"
                      >
                        <td
                          class="ant-table-cell"
                        >
                          hello_world
                        </td>
                        <td
                          class="ant-table-cell"
                        >
                          org_D4ES55BSeeAHystq
                        </td>
                        <td
                          class="ant-table-cell"
                        >
                          <div
                            class="ant-flex css-dev-only-do-not-override-ni1kz0 ant-flex-align-center"
                            style="gap: 5px;"
                          >
                            <span
                              style="font-size: 14px; font-weight: 500; font-style: normal; text-overflow: ellipsis; overflow: hidden;"
                            >
                              helloworld123
                            </span>
                            <span
                              data-testid="mocked-svg"
                              style="display: flex; justify-content: center;"
                            >
                              <svg
                                data-testid="info"
                              />
                            </span>
                          </div>
                        </td>
                        <td
                          class="ant-table-cell"
                        >
                          <div
                            class="ant-flex css-dev-only-do-not-override-ni1kz0"
                            style="gap: 2px;"
                          >
                            <svg
                              data-testid="status"
                            />
                            <span
                              class="ant-typography css-dev-only-do-not-override-ni1kz0"
                            >
                              Active
                            </span>
                          </div>
                        </td>
                        <td
                          class="ant-table-cell"
                        >
                          <div
                            class="ant-space css-dev-only-do-not-override-ni1kz0 ant-space-horizontal ant-space-align-center ant-space-gap-row-small ant-space-gap-col-small"
                          >
                            <div
                              class="ant-space-item"
                            >
                              <button
                                class="ant-btn css-dev-only-do-not-override-ni1kz0 ant-btn-link ant-btn-color-dangerous ant-btn-variant-link"
                                data-testid="handle-modify"
                                style="padding: 4px 0px;"
                                type="button"
                              >
                                <span>
                                  Modify
                                </span>
                              </button>
                            </div>
                            <div
                              class="ant-space-item"
                            >
                              <button
                                class="ant-btn css-dev-only-do-not-override-ni1kz0 ant-btn-link ant-btn-color-dangerous ant-btn-variant-link"
                                data-testid="handle-deactivate"
                                style="padding: 4px 0px;"
                                type="button"
                              >
                                <span>
                                  Deactivate
                                </span>
                              </button>
                            </div>
                          </div>
                        </td>
                      </tr>
                      <tr
                        class="ant-table-row ant-table-row-level-0"
                        data-row-key="org_AeltT2tTQsOFNvCC"
                      >
                        <td
                          class="ant-table-cell"
                        >
                          QE Testing
                        </td>
                        <td
                          class="ant-table-cell"
                        >
                          org_AeltT2tTQsOFNvCC
                        </td>
                        <td
                          class="ant-table-cell"
                        >
                          <div
                            class="ant-flex css-dev-only-do-not-override-ni1kz0 ant-flex-align-center"
                            style="gap: 5px;"
                          >
                            <span
                              style="font-size: 14px; font-weight: 500; font-style: normal; text-overflow: ellipsis; overflow: hidden;"
                            >
                              qetesting
                            </span>
                            <span
                              data-testid="mocked-svg"
                              style="display: flex; justify-content: center;"
                            >
                              <svg
                                data-testid="info"
                              />
                            </span>
                          </div>
                        </td>
                        <td
                          class="ant-table-cell"
                        >
                          <div
                            class="ant-flex css-dev-only-do-not-override-ni1kz0"
                            style="gap: 2px;"
                          >
                            <svg
                              data-testid="status"
                            />
                            <span
                              class="ant-typography css-dev-only-do-not-override-ni1kz0"
                            >
                              Inactive
                            </span>
                          </div>
                        </td>
                        <td
                          class="ant-table-cell"
                        >
                          <div
                            class="ant-space css-dev-only-do-not-override-ni1kz0 ant-space-horizontal ant-space-align-center ant-space-gap-row-small ant-space-gap-col-small"
                          >
                            <div
                              class="ant-space-item"
                            >
                              <button
                                class="ant-btn css-dev-only-do-not-override-ni1kz0 ant-btn-link ant-btn-color-dangerous ant-btn-variant-link"
                                data-testid="handle-activate"
                                style="padding: 4px 0px;"
                                type="button"
                              >
                                <span>
                                  Activate
                                </span>
                              </button>
                            </div>
                          </div>
                        </td>
                      </tr>
                    </tbody>
                  </table>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;
